spyder/plugins/findinfiles/widgets/main_widget.py

```diff
@@ -430,6 +437,8 @@ def resizeEvent(self, event):
         """Adjustments when the widget is resized."""
         super().resizeEvent(event)
 
+        self.search_text_edit.setMaximumWidth(self.width())
```

Comment:
```suggestion
        # This is necessary to prevent the width of search_text_edit to control
        # the width of the pane.
        # Fixes spyder-ide/spyder#24188
        self.search_text_edit.setMaximumWidth(self.width())
```